user-ms/src/main/kotlin/puc/service/impl/UserAppServiceImpl.kt

```diff
@@ -87,4 +92,50 @@ class UserAppServiceImpl (
             refreshToken
         )
     }
+
+    override fun updateUser(id: Long, request: UpdateUserRequest): UserResponse {
+        val user = userRepository.findById(id).orElseThrow { UserNotFoundException(MESSAGE_ERRO_USER_NOT_FOUND) }
+        this.checkUserUniqueness(user, request)
+        this.checkIfRolesAreValid(request.roles)
+        user.email = request.email
```

Comment:
podia ta naquele classe map útil

Reply:
Se for desse jeito eu achei meio esquisito, parece que tá re-instanciando 2 classes (UserReponse & UserApp)  atoa e ficou mais verboso, vou manter do jeito que tá mesmo a não ser que você tenha outra sugestão ou argumento.

![image](https://github.com/user-attachments/assets/ab483c49-f99a-4ed5-86e7-26ccdda8716a)